### PR TITLE
Add more exercises to Configuration episode

### LINKED ISSUE #134: Output of recipe in episode 4 for v2.0.0
The current output (stdout/log files/settings) of running the example recipe in episode 4 uses ESMValTool v2.0.0b9 . It would be nice if the output would reflect the version end users will install so their output and the tutorials output is as similar as possible.

### LINKED ISSUE #148: Improve configuration episode
Currently the configuration episode is just a long list with (textual) explanation of the settings. For interactive (carpentry-style) teaching it would be nicer if the episode consisted of a number of exercises that also facilitate later episodes. For example:

1. Find the default value for `remove_preproc_dir` and set it to True (related to #148)
2a (local machine): Find the config option to specify the location of the input data. Add the data path where you downloaded the tutorial data, OR:
2b (site-specific): Find the config option to specify the location of the input data. Try to find the specific settings for the machine that you're using and substitute those settings.
3. Use the documentation to find out what options are available for the 'DRS' option in the config-user file.

### PR DESCRIPTION
**Pull request by SarahAlidoost**
# Pull Request checklist

We appreciate your time and effort to improve the tutorial. Please keep in mind that lesson maintainers are volunteers and it may be some time before they can respond to your contribution.

* * *

## Before you start

- [x] Read [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md).
- [x] Create an [issue](https://github.com/ESMValGroup/tutorial/issues) to discuss your idea. This allows your contributions to be incorporated into the tutorial.

## Tasks

- [x] Give this pull request a descriptive title.
- [x] If you are contributing to existing lesson materials, please make sure the content conforms to the `Lesson development` section in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) and does not contain any spelling or grammatical errors.
- [ ] If you are making a new episode, please make sure the content conforms to the `Lesson organization` and `Lesson formatting` sections in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) and does not contain any spelling or grammatical errors.
- [ ] Preferably Codacy checks pass. Status can be seen below your pull request. If there is an error, click the link to find out why.
- [x] Preview changes on your machine before pushing them to GitHub by running `make serve`, alternatively `make docker-serve`. Please see the `Previewing your changes locally` section in [CONTRIBUTING.md](https://github.com/ESMValGroup/tutorial/blob/master/CONTRIBUTING.md) for installation instructions.
- [x] All code instructions have been tested.

If you need help with any of the tasks above, please do not hesitate to ask by commenting in the issue or pull request.

* * *

Closes #148

### CODE REVIEW

File: _episodes/03-configuration.md [hunk 1]
```diff
@@ -12,8 +12,9 @@ objectives:
 
 keypoints:
 - The ``config-user.yml`` tells ESMValTool where to find input data.
-- "``rootpath`` defines the root directory for the input data."
 - "``output_dir`` defines the destination directory."
+- "``rootpath`` defines the root path of the data."
+- "``drs`` defines the directory structure of the data."
 
 ---
```
Comment (Peter9192) [inline _episodes/03-configuration.md:20]: I cannot comment on line 26, but I would like to rephrase that. Now, we show a link 'an example file can be found there'. But it's not an example file, it is _the_ default file. So when you type `esmvaltool config get ...`, this is the file that is copied to your home folder. The learners will not need to be able to locate the file in the core package, so why not just change this to: 

"By typing `esmvaltoog config get ...` a copy of the default configuration will be stored in your home directory, where you can then modify it as seen fit."

File: _episodes/03-configuration.md [hunk 2]
```diff
@@ -184,71 +209,131 @@ rootpath:
...
 >
-> - To get the data (or its correct rootpath), check instruction in
->   [Setup]({{ page.root }}{% link setup.md %}).
-> - For more information about setting the rootpath, see also the ESMValTool
->   [documentation](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/esmvalcore/datafinder.html).
-{: .callout}
+> In this tutorial, we will work with data from
+> [CMIP5](https://esgf-node.llnl.gov/projects/cmip5/)
+> and [obs4mips](https://esgf-node.llnl.gov/projects/obs4mips/).
+> How does ESMValTool find the data?
```
Comment (Peter9192) [inline _episodes/03-configuration.md:221]: Very nice that this is now an exercise! I would rephrase this to something more 'active':
```suggestion
> Modify the rootpath section of the tutorial to make sure the data path for the tutorial is set correctly for both CMIP5 and obs4mips. 
```

File: _episodes/03-configuration.md [hunk 3]
```diff
@@ -184,71 +209,131 @@ rootpath:
...
-> (e.g. ``BADC``, ``CP4CDS``, ``DKRZ``, ``ETHZ``, ``SMHI``, ``BSC``). For more
-> information about ``drs``, you can visit the ESMValTool
-> [documentation](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/quickstart/find_data.html#cmor-drs).
-{: .callout}
-
-> ## Make your own configuration file
+> In this lesson, we will work with data from
+> [CMIP5](https://esgf-node.llnl.gov/projects/cmip5/)
+> and [obs4mips](https://esgf-node.llnl.gov/projects/obs4mips/).
+> How does ESMValTool know the data structure?
```
Comment (Peter9192) [inline _episodes/03-configuration.md:279]:
```suggestion
> Set the correct `drs` for the tutorial. If you're working on your local machine, use the 'default'. Otherwise, find the correct drs specification for your cluster.
```

File: _episodes/03-configuration.md [hunk 4]
```diff
@@ -184,71 +209,131 @@ rootpath:
...
+>> Therefore the default directory must be structured accordingly with sub-directories
+>> `TierX` e.g. Tier1, Tier2 or Tier3, even when `drs: default`.
+>>
 > {: .solution}
 {: .challenge}
 
+> ## Make your own configuration file
+>
+> It is possible to have several configuration files with different purposes,
+> for example: config-user_formalised_runs.yml, config-user_debugging.yml
```
Comment (Peter9192) [inline _episodes/03-configuration.md:336]:
```suggestion
> for example: config-user_formalised_runs.yml, config-user_debugging.yml. In this case, you have to pass the path to the config file you want to use as a command line option when running the ESMValTool.
```
ESMValTool will first look for the default file (first in your ~/.esmvaltool dir, then in the installation dir). Only if you manually specify another path it will use that one. (This has recently changed. Before, you _always_ had to add the path).

**Review by Peter9192 — CHANGES_REQUESTED** [review]
Hey @SarahAlidoost ,
This is already a huge improvement, thanks for working on this! 

For this PR to fully address #148, I suggested a couple of additional changes that I think would improve the episode even further. Could you have another look at those?

File: _episodes/03-configuration.md [hunk 5]
```diff
@@ -98,7 +99,22 @@ config_developer_file: null
 profile_diagnostic: false
```
[the remark this answers is not in the document]
Comment (SarahAlidoost) [reply _episodes/03-configuration.md:99]: Agree.

File: _episodes/03-configuration.md [hunk 6]
```diff
@@ -130,6 +140,22 @@ are not plots, e.g. files in NetCDF format (depends on the diagnostic script).
 [lesson]({{ page.root }}{% link _episodes/04-recipe.md %})
 {: .callout}
 
+> ## Set the destination directory
+>
+> Let's name our destination directory ``esmvaltool_output`` in the working directory.
+> How to tell ESMValTool about our destination directory?
```
[the remark this answers is not in the document]
Comment (SarahAlidoost) [reply _episodes/03-configuration.md:146]: @Peter9192 thanks for mentioning the handbook. Action verbs are useful in developing [learning objectives ](https://carpentries.github.io/curriculum-development/developing-content.html), section 5.2.1. I didn't find any recommendation using an action verb for [designing challenges](https://carpentries.github.io/curriculum-development/designing-challenges.html). However, I have looked at many[ carpentry lessons ](https://software-carpentry.org/lessons/) to figure out how to design a challenge. Of course, I can revise that sentence a bit following your suggestion without using an action verb.

File: _episodes/03-configuration.md [hunk 7]
```diff
@@ -130,6 +140,22 @@ are not plots, e.g. files in NetCDF format (depends on the diagnostic script).
...
+>> We use `output_dir` entry in the `config-user.yml` file as:
+>>```yaml
+>> output_dir: ./esmvaltool_output
+>>```
+>>
+>> If the `esmvaltool_output` does not exist, ESMValTool will generate it for you.
+> {: .solution}
+{: .challenge}
+
 ## Auxiliary data directory
```
[the remark this answers is not in the document]
Comment (SarahAlidoost) [reply _episodes/03-configuration.md:159]: I made it as a call-out. Then, it goes under the previous section. To prevent this I made a new section  "Other settings" at the end of this lesson.

File: _episodes/03-configuration.md [hunk 8]
```diff
@@ -164,7 +189,6 @@ Using the number there you can increase the number of parallel tasks
 again to a reasonable number for the amount of memory available in your system.
 {: .callout}
```
[the remark this answers is not in the document]
Comment (SarahAlidoost) [reply _episodes/03-configuration.md:190]: I made it as a call-out. Then, it goes under the previous section. To prevent this I made a new section  "Other settings" at the end of this lesson.

File: _episodes/03-configuration.md (hunk 2 above)
Comment (SarahAlidoost) [reply _episodes/03-configuration.md:221]: It is revised.

File: _episodes/03-configuration.md (hunk 3 above)
Comment (SarahAlidoost) [reply _episodes/03-configuration.md:279]: You are giving the answer to the challenge in this line. I revised the question a bit without mentioning the answer.

File: _episodes/03-configuration.md [hunk 9]
```diff
@@ -98,7 +99,22 @@ config_developer_file: null
 profile_diagnostic: false
 ```
 
-In general there is no need to change the settings listed above.
+> ## Saving preprocessed data
+>
+> In the configuration file, which settings are useful to make sure preprocessed
```
[the remark this answers is not in the document]
Comment (SarahAlidoost) [reply _episodes/03-configuration.md:104]: it is revised.

File: _episodes/03-configuration.md (hunk 1 above)
Comment (SarahAlidoost) [reply _episodes/03-configuration.md:20]: it is revised.

**Comment by SarahAlidoost** [comment]
@Peter9192 thank you for the comments, very helpful. Please see the new revisions.

File: _episodes/03-configuration.md [hunk 10]
```diff
@@ -65,40 +65,26 @@ This file contains the information for:
 
 These settings are used to inform ESMValTool about your preference about
```
Comment (Peter9192) [inline _episodes/03-configuration.md:66]: This sentence, referring to "these settings" is a bit out of the blue now. Can we rephrase that?

**Review by Peter9192 — CHANGES_REQUESTED** [review]
Hey @SarahAlidoost great work! Thank you very much. 
Just one more small suggestion.

File: _episodes/03-configuration.md (hunk 10 above)
Comment (SarahAlidoost) [reply _episodes/03-configuration.md]: Sure we can, something like this?

Comment (Peter9192) [reply _episodes/03-configuration.md]: Great, thanks!

**Review by Peter9192 — APPROVED** [review]
